src/compiler/sys.ts

```diff
@@ -500,7 +500,10 @@ namespace ts {
                     for (const current of files) {
                         const name = combinePaths(path, current);
                         if (!contains(exclude, getCanonicalPath(name))) {
-                            const stat = _fs.statSync(name);
+                            // fs.statSync would throw an exception if the file is a symlink
+                            // whose linked file doesn't exist. fs.lstatSync would return a stat
+                            // object for the symlink file itself in this case
+                            const stat = _fs.lstatSync(name);

```

Comment:
This was an issue that caused tsserver to crash when editing the repro sample. The ember server generates tons of symlink files in the `tmp` folder, and then deletes the linked files after the server shuts down. Then these symlink files became invalid ones, which caused `fs.statSync` to throw exceptions when we try to scan the folder.

The alternative solution was to try `fs.statSync` first, if an exception was thrown just skip this file.